python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_project.py

```diff
@@ -9,13 +12,22 @@
 
 def test_expected_but_missing() -> None:
     with copy_directory(test_jaffle_shop_path) as project_dir:
-        dbt_artifacts = DbtProject(project_dir=project_dir)
+        project = DbtProject(project_dir=project_dir)
 
         with pytest.raises(DagsterDbtManifestNotFoundError):
-            _ = dbt_artifacts.manifest_path
+            _ = project.manifest_path
 
-        dbt_artifacts.prepare_for_deployment()
-        assert dbt_artifacts.manifest_path.exists()
+        project.prepare_for_deployment()
+        assert project.manifest_path.exists()
+
+        @dbt_assets(manifest=project.manifest_path)
+        def my_assets(context: AssetExecutionContext, dbt: DbtCliResource): ...
+
+        defs = Definitions(
+            assets=[my_assets],
+            resources={"dbt": DbtCliResource.from_project(project)},
```

Comment:
thoughts? kinda seems net neutral to me given all the current callsites have to do `os.fspath`

Reply:
the only way to take a direct arg that isnt config on a `ConfigurableResource` is `PrivateAttr()`with a `_` prefix which seems worse to me

Reply:
@benpankow is there an easy path to:
- Omit an argument on `ConfigurableResource` from being rendered as Dagster configuration, and 
- Still continue to let that argument be accessible as a public property on the object?

Reply:
There's a hacky way which is to treat the param as a nested resource - basically every constructor arg must either be a resource (e.g. any python object) or config (shows up in the config editor)/

```py
from dagster import ConfigurableResource, Definitions, ResourceDependency, asset


class MyConfigurableResource(ConfigurableResource):
    a_str: str
    a_hidden_str_param: ResourceDependency[str]

    def print_my_strings(self):
        print(self.a_str, self.a_hidden_str_param)


@asset
def my_asset(my_resource: MyConfigurableResource):
    my_resource.print_my_strings()


defs = Definitions(
    assets=[my_asset],
    resources={
        "my_resource": MyConfigurableResource(a_str="foo", a_hidden_str_param="bar")
    },
)

```


<img width="217" alt="Screenshot 2024-03-22 at 4 23 00 PM" src="https://github.com/dagster-io/dagster/assets/10215173/e481ccc2-7562-4ff3-a7e0-9e59478e1ef5">
<img width="293" alt="Screenshot 2024-03-22 at 4 23 10 PM" src="https://github.com/dagster-io/dagster/assets/10215173/d7da1474-9fc4-4274-8de2-35b7fc2cbd93">

Reply:
@benpankow curious for you to take a look at the current override `__init__` approach and assess if you think the trick you presented here would be better - it doesn't seem that way to me but I can't say i fully wrap my head around it.